File: QadeerApp.Web/Modules/Administration/TrainingTerm/TrainingTermColumns.cs
namespace QadeerApp.Administration.Columns;

[ColumnsScript("Administration.TrainingTerm")]
[BasedOnRow(typeof(TrainingTermRow), CheckNames = true)]
public class TrainingTermColumns
{
    [EditLink, AlignRight, Width(60)]
    public int TrainingTermId { get; set; }

    [EditLink, Width(200)]
    public string Name { get; set; }

    [Width(80), QuickFilter]
    public short IsActive { get; set; }
}

File: QadeerApp.Web/Modules/ServerTypes/Administration/TrainingTermColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { TrainingTermRow } from "./TrainingTermRow";

export interface TrainingTermColumns {
    TrainingTermId: Column<TrainingTermRow>;
    Name: Column<TrainingTermRow>;
    IsActive: Column<TrainingTermRow>;
}

export class TrainingTermColumns extends ColumnsBase<TrainingTermRow> {
    static readonly columnsKey = 'Administration.TrainingTerm';
    static readonly Fields = fieldsProxy<TrainingTermColumns>();
}
